pages/docs/modules/exec.mdx

```diff
@@ -0,0 +1,355 @@
+# exec
+
+## Functions
+
+### command
+
+```go filename="Function signature"
+command(file string, args ...string) exec.command
+```
+
+Creates a new `exec.command` targeting a specific `file` to execute,
+followed by arguments to execute the command with.
+
+Nothing is executed by this function alone, but instead only a [`exec.command`](#type-command)
+object is returned. To execute `file`, you must call the [`run`](#commandrun)
+method on the [`exec.command`](#type-command) object.
+
+The lookup if `file` executable exists is also not performed until the [`run`](#commandrun)
+method is called.
+
+```go copy filename="Example"
+>>> exec.command("bash")
+exec.command("bash")
+>>> exec.command("bash", "-c", "echo hello world")
+exec.command("bash", "-c", "echo hello world")
+>>> exec.command("file-that-does-not-exist")
+exec.command("file-that-does-not-exist")
+>>> cmd := exec.command("date", "-Iseconds")
+>>> cmd.run()
+>>> cmd.stdout
+"2024-01-09T21:33:48+01:00\n"
+```
+
+### look_path
+
+```go filename="Function signature"
+look_path(file string) string
+```
+
+Searches for an executable named `file` in the directories named by the `PATH`
+environment variable.
+If `file` contains a slash, it is tried directly and the `PATH` is not consulted.
+Otherwise, on success, the result is an absolute path.
+
+```go copy filename="Example"
+>>> exec.look_path("bash")
+"/usr/bin/bash"
+```
+
+### exec
+
+```go filename="Function signature"
+exec(file string, args list = [], params map = {}) exec.result
```

Comment:
This is the first function with optional arguments (that I can find) in the documentation.

Is this a valid syntax, or should we perhaps go with "function overloads" approach?

```suggestion
exec(file string) exec.result
exec(file string, args list) exec.result
exec(file string, args list, params map) exec.result
```

Or what about instead skipping the types? I don't like this as much, as the types are still valuable info, but maybe the default values give sufficient info?

```suggestion
exec(file string, args=[], params={}) exec.result
```

Reply:
Yeah, I agree this is all tricky. Generally I think showing the optional values in the signature makes sense. I do think the variant that skips the types looks clean.